src/IntentBuilder.ts

```diff
@@ -189,50 +211,34 @@ export class IntentBuilder {
    * @returns The determined target state.
    */
   private setTo(state: State): ToState {
-    switch (true) {
-      case state instanceof Asset:
-        return { case: 'toAsset', value: state };
-      case state instanceof Stake:
-        return { case: 'toStake', value: state };
-      case state instanceof Loan:
-        return { case: 'toLoan', value: state };
-      default:
-        return { case: undefined };
-    }
+    if (state instanceof Asset) return { case: 'toAsset', value: state };
+    if (state instanceof Stake) return { case: 'toStake', value: state };
+    if (state instanceof Loan) return { case: 'toLoan', value: state };
+    return { case: undefined };
   }
 
   /**
-   * Signs a transaction using the account details and transaction builder.
-   * @param chainId The chain ID where the signature is applicable.
-   * @param account The account performing the signing.
-   * @param builder The UserOperationBuilder containing transaction details.
-   * @returns The signature as a string.
+   * Creates a UserOperationBuilder instance with the specified parameters.
+   * @param chainId The chain ID for the UserOperation.
+   * @param account The user account associated with the UserOperation.
+   * @param calldata The calldata for the UserOperation.
+   * @param opts Optional UserOperation options.
+   * @returns A promise that resolves to a UserOperationBuilder instance.
    */
-  private async sign(chainId: number, account: Account, builder: UserOperationBuilder) {
-    const userOp = builder.getOp();
-    const packedData = ethers.AbiCoder.defaultAbiCoder().encode(
-      ['address', 'uint256', 'bytes32', 'bytes32', 'uint256', 'uint256', 'uint256', 'uint256', 'uint256', 'bytes32'],
-      [
-        userOp.sender,
-        userOp.nonce.toString(),
-        ethers.keccak256(userOp.initCode as BytesLike),
-        ethers.keccak256(userOp.callData as BytesLike),
-        userOp.callGasLimit.toString(),
-        userOp.verificationGasLimit.toString(),
-        userOp.preVerificationGas.toString(),
-        userOp.maxFeePerGas.toString(),
-        userOp.maxPriorityFeePerGas.toString(),
-        ethers.keccak256(userOp.paymasterAndData as BytesLike),
-      ],
-    );
-
-    const enc = ethers.AbiCoder.defaultAbiCoder().encode(
-      ['bytes32', 'address', 'uint256'],
-      [ethers.keccak256(packedData), ENTRY_POINT, chainId],
-    );
-
-    const userOpHash = ethers.keccak256(enc);
-    return await account.signer.signMessage(ethers.getBytes(userOpHash));
+  private async createUserOpBuilder(
+    chainId: number,
```

Comment:
why do we need both createUserOpBuilder() and userOpBuilder()?

Reply:
to have flexibility to pass calldata (opts can't be optional)